### PR TITLE
[IA-2234]  use sam api/resource/v2

### PR DESCRIPTION
**Pull request by jdcanas**
I tested this by calling the leo APIs that call these APIs under the covers and everything looked good.

### CODE REVIEW

**Comment by jdcanas** [comment]
jenkins retest

**Comment by jdcanas** [comment]
jenkins retest

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala [hunk 1]
```diff
@@ -393,12 +393,21 @@ object HttpSamDAO {
       email <- x.downField("email").as[SamPolicyEmail]
     } yield SyncStatusResponse(lastSyncDate, email)
   }
+
+  final case class SamRoleAction(roles: List[SamPolicyName])
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala]: Can you put this near the other case classes?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala [hunk 2]
```diff
@@ -417,7 +426,7 @@ object HttpSamDAO {
 
 final case class CreateSamResourceRequest[R](samResourceId: R, policies: Map[SamPolicyName, SamPolicyData])
 final case class SyncStatusResponse(lastSyncDate: String, email: SamPolicyEmail)
-final case class ListResourceResponse[R](samResourceId: R, samPolicyName: SamPolicyName)
+final case class ListResourceResponse[R](samResourceId: R, samPolicyName: Set[SamPolicyName])
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala]: Nit: `samPolicyNames: Set[SamPolicyName]`

**Comment by rtitle** [comment]
Couple nits, but overall looks correct based on my understanding of the new API

**Review by rtitle — APPROVED** [review]

**Comment by jdcanas** [comment]
jenkins retest

**Pull request merged**